BackupDatabase-Recipe.rb

```diff
@@ -0,0 +1,76 @@
+require 'nokogiri'
+
+def get_database_credentials_for_magento
+  configXMLFile = Nokogiri::XML(File.open("#{current_release}#{app_path}/app/etc/local.xml"))
+  dbCredentials = Hash.new
+  ["host", "username", "password", "dbname"].each do |credential|
+    dbCredentials[credential] = configXMLFile.xpath("//config//global//resources//default_setup//#{credential}/text()").text
+  end
+  dbCredentials
+end
+
+def get_database_credentials_for_cakephp
+  configFile = File.open("#{current_release}#{app_path}/Config/database.php").read
+
+  rowCredentials= configFile.scan(/'(?<key>\w+)'\s=>\s'(?<value>\w+)'/)
+  dbCredentials = Hash.new
+
+  mapping = {"login" => "username", "database" => "dbname"}
+  rowCredentials.each do |key,value|
+    credentialKey = mapping[key].nil? ? key : mapping[key]
+    dbCredentials[credentialKey] = value
+  end
+
+  dbCredentials
+end
+
+def is_cakephp_project
+  exists?(:cake_migrations)
+end
+
+namespace :database do
+
+  desc <<-DESC
+		Generate a backup for database project.
+  DESC
+  task :backup, :roles => :db, :only => { :primary => true } do
+    filename = "#{application}.dump.sql.gz"
+    file = "/tmp/#{filename}"
+    on_rollback { delete file }
+
+    logger.info("Fetching database credentials from project")
+    dbCredentials = is_cakephp_project ? get_database_credentials_for_cakephp : get_database_credentials_for_magento
```

Comment:
It would be better to have credentials as a requirement fr using the recipe (global variable, or injected dependency), so extension to any technology would be possible, and vendor specific logic could be moved to appropriate recipe. It is also more futureproof (for instance Magento2, CakePHP3 ...)

For instance the `deploy.rb` file would contain something like: `set :dbCredentials, Cake::dbCredentials()`.